src/muse/mca.py

```diff
@@ -538,94 +538,44 @@ def find_equilibrium(
 
     from numpy import ones
 
-    from muse.utilities import future_propagation
-
     market = market.copy(deep=True)
     if excluded_commodities:
         included = ~market.commodity.isin(excluded_commodities)
     else:
         included = ones(len(market.commodity), dtype=bool)
 
-    market["updated_prices"] = drop_timeslice(market.prices.copy())
     prior_market = market.copy(deep=True)
     converged = False
-    equilibrium_sectors = sectors
-    for i in range(maxiter):
-        market["prices"] = drop_timeslice(market.updated_prices)
+    iteration = 0
+    while iteration < maxiter and not converged:
         prior_market, market = market, prior_market
         market.consumption[:] = 0.0
         market.supply[:] = 0.0
         market, equilibrium_sectors = single_year_iteration(market, sectors)
 
-        check_demand_fulfillment(market.sel(commodity=included), tol_unmet_demand)
+        if maxiter == 1 or not equilibrium:
+            converged = True
+            break
```

Comment:
So this means that if the maximum number of iterations is 1, then we treat it as if it's always converged. Is that right? Does `maxiter == 1` have a special meaning?

Reply:
Yes, `maxiter = 1` is treated the same as `equilibrium = False`. See [here](https://muse-os.readthedocs.io/en/latest/inputs/projections.html#initial-market-projection)

Reply:
Ah ok. Can we be sure that the user hasn't set `equilibrium = True` and `maxiter = 1` here?

Reply:
I guess in that case the `maxiter=1` will win out. This is sort of explained in the documentation (albeit subtly):

"If the MCA works in an exogenous mode (setting market keyword equilibrium = false or maximum_iterations = 1), it will use the initial market projection as the projection for the the base year and all the future periods of the simulation"

The key word being "or"